### PR TITLE
[jvm-packages] Bump rapids version.

### PR DESCRIPTION
**Pull request by trivialfis**
Close https://github.com/dmlc/xgboost/pull/8559 .

### CODE REVIEW

File: jvm-packages/xgboost4j-spark-gpu/src/main/scala/ml/dmlc/xgboost4j/scala/rapids/spark/GpuUtils.scala [hunk 1]
```diff
@@ -31,16 +31,7 @@ private[spark] object GpuUtils {
...
-      val clazz = Utils.classForName("com.nvidia.spark.rapids.GpuColumnVectorUtils")
-      clazz.getDeclaredMethod("extractHostColumns", classOf[Table], classOf[Array[DataType]])
-        .invoke(null, table, types).asInstanceOf[Array[ColumnVector]]
-    } catch {
-      case _: ClassNotFoundException =>
-        // If it's older version, use the GpuColumnVector
-        GpuColumnVector.extractColumns(table, types).map(_.copyToHost())
-      case e: Throwable => throw e
-    }
+    GpuColumnVectorUtils.extractHostColumns(table, types)
```
Comment (hcho3) [inline jvm-packages/xgboost4j-spark-gpu/src/main/scala/ml/dmlc/xgboost4j/scala/rapids/spark/GpuUtils.scala:34]: We can now remove the workaround introduced from #8240

**Comment by hcho3** [comment]
@wbo4958 Can you look at this error?

> [ERROR] /workspace/jvm-packages/xgboost4j-spark-gpu/src/test/scala/ml/dmlc/xgboost4j/scala/rapids/spark/GpuTestSuite.scala:29: object TrampolineUtil is not a member of package org.apache.spark.sql.rapids.execution
> [ERROR] /workspace/jvm-packages/xgboost4j-spark-gpu/src/test/scala/ml/dmlc/xgboost4j/scala/rapids/spark/GpuTestSuite.scala:235: not found: value TrampolineUtil

The error occurs at runtime, even though the `org.apache.spark.sql.rapids.execution` package has `TrampolineUtil`: https://www.javadoc.io/doc/com.nvidia/rapids-4-spark_2.12/latest/org/apache/spark/sql/rapids/execution/TrampolineUtil$.html

**Comment by trivialfis** [comment]
I will come back to this later if no one takes up. I'm using WSL for work at the moment .......

**Comment by wbo4958** [comment]
I will check it

File: jvm-packages/xgboost4j-spark-gpu/src/main/scala/ml/dmlc/xgboost4j/scala/rapids/spark/GpuUtils.scala (hunk 1 above)
Comment (wbo4958) [reply jvm-packages/xgboost4j-spark-gpu/src/main/scala/ml/dmlc/xgboost4j/scala/rapids/spark/GpuUtils.scala:34]: Please don't remove this workaround for this time, or else xgboost can't work well with older spark-rapids (before 22.10). Let's remove it for the next release.

**Comment by wbo4958** [comment]
@hcho3 could we hold on this PR which will cause the xgboost can't run with the older rapids packages.

**Comment by hcho3** [comment]
@wbo4958 XGBoost is currently using a very old version of Rapids (21.08.2). Wouldn't it cause issues when someone tries to use XGBoost with the latest Rapids?

**Comment by wbo4958** [comment]
> @wbo4958 XGBoost is currently using a very old version of Rapids (21.08.2). Wouldn't it cause issues when someone tries to use XGBoost with the latest Rapids?

@hcho3, Yeah, actually, we have an internal CI pipeline to test the latest XGBoost GPU JVM  + latest Rapids. If there are some issues, I will fix them.

**Comment by hcho3** [comment]
@wbo4958 Interesting, so it's okay if we had two different versions of Rapids running concurrently?

**Comment by wbo4958** [comment]
@hcho3, Yeah, you're right. the rapids version xgboost depends on is quite old. I will fix it. Let's introduce this breaking change in this version.

**Comment by wbo4958** [comment]
> @wbo4958 Interesting, so it's okay if we had two different versions of Rapids running concurrently?

Spark will pick one to run.

**Comment by wbo4958** [comment]
@hcho3 @trivialfis I filed another PR https://github.com/dmlc/xgboost/pull/8648 to bump rapids version to the latest 22.12.0

**Comment by wbo4958** [comment]
@hcho3 I guess we can close this PR since the alternative PR #https://github.com/dmlc/xgboost/pull/8648 is merged.

**Pull request closed without merging**